### ISSUE TITLE
Duplicated object key in const not detected when the key is a number preceded by `-` or `+`

### ISSUE THREAD

**Issue by rigwild** [microsoft/TypeScript#38295]
**TypeScript Version:**  `3.9.0-dev.20200422`

**Search Terms:**

 - Signed numbers
 - Object keys negative
 - readonly negative key
 - object key negative
 - object key duplicate
 - as const negative
 - object key operand
 - others...

**Code**

```ts
export const rotate = {
  "-45": "test",
  [-45]: "test" // Duplicate NOT detected
};

export const rotate2 = {
  "45": "test",
  [+45]: "test" // Duplicate NOT detected
};

export const rotate4 = {
  "45": "test",
  [45]: "test" // Duplicate detected
};
```

**Expected behavior:**
TypeScript should alert the user that there's a duplicated key.

**Actual behavior:**
Key duplication is not detected when the key is a number preceded by `-` or `+`.

[Playground Link](https://www.typescriptlang.org/play?ts=4.0.0-dev.20200501#code/KYDwDg9gTgLgBAYwgOwM7yhGBDGw4C8cA3gLABQccARALQAsArNQFw17rUA0FVA2g0YBdNtQ4xqcAPRS4AEQCuYADYBLBLnwA5APIAVOABNgeBHkMUAvgG4KFUJFiIU6OJhx4ATIRK8aTVnZgTh5KOD4AaiYRIM5pWUUVdU04XQNjU3MrW3J7cGh4JDQMLE16HzIw6gDRcW4-Pmja4Il4+SU1DTwjE2AzYAtyGyA)

**Related Issues:** 
#37866 #38133 #37292

The bug seems to be there for a very long time (it has the same behavior on v2.4.1 on the playground).

**Comment by lissein** [comment]
Hi, I'll try to work on this.
